features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -28,28 +29,43 @@ using namespace mbed;
 struct at_reg_t {
     const CellularNetwork::RegistrationType type;
     const char *const cmd;
+    const char *const urc_prefix;
 };
 
 static const at_reg_t at_reg[] = {
-    { CellularNetwork::C_EREG, "AT+CEREG" },
-    { CellularNetwork::C_GREG, "AT+CGREG" },
-    { CellularNetwork::C_REG,  "AT+CREG"  },
+    { CellularNetwork::C_EREG, "AT+CEREG", "+CEREG:"},
+    { CellularNetwork::C_GREG, "AT+CGREG", "+CGREG:"},
+    { CellularNetwork::C_REG,  "AT+CREG", "+CREG:"}
 };
 
 AT_CellularNetwork::AT_CellularNetwork(ATHandler &atHandler) : AT_CellularBase(atHandler),
     _stack(NULL), _apn(NULL), _uname(NULL), _pwd(NULL), _ip_stack_type_requested(DEFAULT_STACK), _ip_stack_type(DEFAULT_STACK), _cid(-1),
-    _connection_status_cb(NULL), _op_act(operator_t::RAT_UNKNOWN), _authentication_type(CHAP), _last_reg_type(C_REG),
-    _connect_status(NSAPI_STATUS_DISCONNECTED), _new_context_set(false)
+    _connection_status_cb(NULL), _op_act(RAT_UNKNOWN), _authentication_type(CHAP), _last_reg_type(C_REG),
+    _connect_status(NSAPI_STATUS_DISCONNECTED), _new_context_set(false), _reg_status(NotRegistered)
 {
-
-    _at.set_urc_handler("NO CARRIER", callback(this, &AT_CellularNetwork::urc_no_carrier));
+    _urc_funcs[C_EREG] = callback(this, &AT_CellularNetwork::urc_cereg);
+    _urc_funcs[C_GREG] = callback(this, &AT_CellularNetwork::urc_cgreg);
+    _urc_funcs[C_REG] = callback(this, &AT_CellularNetwork::urc_creg);
 }
 
 AT_CellularNetwork::~AT_CellularNetwork()
 {
     free_credentials();
 }
 
+nsapi_error_t AT_CellularNetwork::init()
+{
+    for (int type = 0; type < CellularNetwork::C_MAX; type++) {
+        if (has_registration((RegistrationType)type)) {
+            if (_at.set_urc_handler(at_reg[type].urc_prefix, _urc_funcs[type])) {
```

Comment:
could be easier to read with != NSAPI_ERROR_OK

Reply:
true, fixed